go/enclave/components/batch_executor.go

```diff
@@ -183,30 +165,36 @@ func (executor *batchExecutor) ComputeBatch(context *BatchExecutionContext, fail
 	crossChainTransactions := executor.crossChainProcessors.Local.CreateSyntheticTransactions(messages, stateDB)
 	executor.crossChainProcessors.Local.ExecuteValueTransfers(transfers, stateDB)
 
-	transactionsToProcess, freeTransactions := executor.payL1Fees(stateDB, context)
+	transactionsToProcess, freeTransactions := executor.estimateL1Fees(stateDB, context)
+
+	xchainTxs := make(common.L2PricedTransactions, 0)
+	for _, xTx := range crossChainTransactions {
+		xchainTxs = append(xchainTxs, common.L2PricedTransaction{
+			Tx:             xTx,
+			PublishingCost: big.NewInt(0),
+		})
+	}
 
-	crossChainTransactions = append(crossChainTransactions, freeTransactions...)
+	xchainTxs = append(xchainTxs, freeTransactions...)
```

Comment:
shouldn't this be logically the other way round? "freeTransactions" is the array that contains xChainTxs and other stuff.
I think these variables need more clear names and comments